sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -200,8 +200,30 @@ private static string DetermineValidPathCharacters()
         /// <returns></returns>
         public static string GetExtension(string path)
         {
-            if (path == null)
+            if (path is null)
                 return null;
+
+#if NET8_0_OR_GREATER
+            // LastIndexOf and LastIndexOfAny is vectorized on .NET8+ and is
+            // signifigantly faster for cases where 'path' does not end with a short file
+            // extension, such as GUIDs
+            int extensionIndex = path.AsSpan().LastIndexOf('.');
+            if (extensionIndex == -1)
+            {
+                return string.Empty;
+            }
+
+            int directoryIndex = path.AsSpan().LastIndexOfAny('/', '\\', ':');
+
+            // extension separator is found and exists before path separator or path separator doesn't exist
+            // AND it's not the last one in the string
+            if (directoryIndex < extensionIndex && extensionIndex < path.Length - 1)
+            {
+                return path.Substring(extensionIndex);
```

Comment:
If we had the span assigned to a local like suggested above we could slice the span here directly